### PR TITLE
Generate csv for users trace report

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to #452

If you want to add some conditions for getting any of the columns plz tell me, you can see an example of generated CSV

https://docs.google.com/spreadsheets/d/1HmHX4hncPVbwi-VTYpJo_xY0NFjlbYaKmm1r7t9JvqQ/edit?usp=sharing

### CODE REVIEW

**Comment by aminlatifi** [comment]
> related to #452
> 
> If you want to add some conditions for getting any of the columns plz tell me, you can see an example of generated CSV
> 
> https://docs.google.com/spreadsheets/d/1HmHX4hncPVbwi-VTYpJo_xY0NFjlbYaKmm1r7t9JvqQ/edit?usp=sharing

Thank you @mohammadranjbarz 

@bowensanders Is it good enough? there is a point... in this report `# of Campaigns created` counts number of campaigns a user is owner of! for instance I have not created Giveth Dapp Development Campaign and ownership has been transfered to me, but it's counted under my name! In case you want the original creator we should implement it in another way

**Comment by mohammadranjbarz** [comment]
@aminlatifi 

I made changes that you said in https://github.com/Giveth/feathers-giveth/issues/452#issuecomment-810922459

and this is output https://docs.google.com/spreadsheets/d/1X3ACTDkxFbUKtaibw4SCsVSsBepWOt6r5fvQVuv-dX0/edit?usp=sharing

(As you assumed the transactions collection was not complete)

**Comment by aminlatifi** [comment]
We should fill transaction collection with all txs related to giveth trace, then obtain first and last usage of users

**Comment by mohammadranjbarz** [comment]
> We should fill transaction collection with all txs related to giveth trace, then obtain first and last usage of users

I implemend a script for this purpose in last commit of this PR

https://github.com/Giveth/feathers-giveth/pull/457/commits/5adf4303f215a130ebb166e5dbc918141b52e195

**Comment by mohammadranjbarz** [comment]
@aminlatifi 
it's implemented last week

**Comment by aminlatifi** [comment]
@mohammadranjbarz Would you please send an updated output here

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz Would you please send an updated output here

@aminlatifi 
https://docs.google.com/spreadsheets/d/1iBtP7gkNN3VDZTLn65roA4ByhFVGxz6I5zaWP3HpvzU/edit?usp=sharing

**Comment by aminlatifi** [comment]
@mohammadranjbarz We discussed about filling transaction table to get full result on `first use` and `last use`

File: scripts/generateUsersActivityReportCsv.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+      data[key] = val;
+    },
+  };
+};
+
+const app = appFactory();
+app.set('mongooseClient', mongoose);
+const aggregteQuery = [
+  {
+    $match: {},
```
Comment (aminlatifi) [inline scripts/generateUsersActivityReportCsv.js:26]: You can change it to giverId not equal zero

Comment (aminlatifi) [reply scripts/generateUsersActivityReportCsv.js:26]: For instance this user just has signed in and didn't create a giver in smart contract

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: scripts/generateUsersActivityReportCsv.js (hunk 1 above)
Comment (mohammadranjbarz) [reply scripts/generateUsersActivityReportCsv.js:26]: Are you sure ?
for example check below user it has a lot of activity but his giverId is zero 
Screen Shot 1400-01-27 at 08 52 26


https://feathers.beta.giveth.io/users/0x51e37e3716c0ED418a90e519CeEE395d8f6deCB9

But I changed to check the users that have `giverId` field and it has better result,
but if you think `giverId:0` still is ok tell me to add it

File: scripts/generateUsersActivityReportCsv.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  };
+};
+
+const app = appFactory();
+app.set('mongooseClient', mongoose);
+const aggregteQuery = [
+  {
+    $match: {
+      giverId: {
+        $exists: true,
```
Comment (aminlatifi) [inline scripts/generateUsersActivityReportCsv.js:28]: This condition will show users with giverId of zero. Like this one
https://feathers.beta.giveth.io/users/0x97B30e465eB68a4a3333D2142a386D7e20DbE195

This kind of users should be ignored.

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: scripts/generateUsersActivityReportCsv.js (hunk 2 above)
Comment (mohammadranjbarz) [reply scripts/generateUsersActivityReportCsv.js:28]: yeah but some users with `giverId:0` have activity indeed, like this one
https://feathers.beta.giveth.io/users/0x51e37e3716c0ED418a90e519CeEE395d8f6deCB9

Comment (aminlatifi) [reply scripts/generateUsersActivityReportCsv.js:28]: You are right

Comment (aminlatifi) [reply scripts/generateUsersActivityReportCsv.js:28]: @mohammadranjbarz so this one is the latest output?
https://docs.google.com/spreadsheets/d/1tIDLibzAF3g7dsilEH_yakuExbXHoWUNZGdcG0Lez4k/edit?usp=sharing

Comment (mohammadranjbarz) [reply scripts/generateUsersActivityReportCsv.js:28]: @aminlatifi 
Yes it's the last one, 
but if you want some changes I think it's faster to apply the changes in a 1-1 meeting

Comment (aminlatifi) [reply scripts/generateUsersActivityReportCsv.js:28]: > @mohammadranjbarz so this one is the latest output?
> https://docs.google.com/spreadsheets/d/1tIDLibzAF3g7dsilEH_yakuExbXHoWUNZGdcG0Lez4k/edit?usp=sharing

@willyogo @bowensanders You can see the latest version in this link

**Pull request merged**